### ISSUE TITLE
tf.keras.layers.MaxPool1D causes CPU floating point exception with invalid pooling window size

### ISSUE THREAD

**Issue by TTZTTZ1** [tensorflow/tensorflow#125509]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 22.04.5 LTS

### Mobile device

N/A

### Python version

3.10.20

### Bazel version

N/A

### GCC/compiler version

N/A

### CUDA/cuDNN version

CUDA 13.0.88 cuDNN 9.x

### GPU model and memory

NVIDIA GeForce RTX 5090 32G

### Current behavior?

`tf.keras.layers.MaxPool1D` causes inconsistent behavior between CPU and GPU when the pooling window size is invalid.

Using a `pool_size` larger than the input length with `padding="valid"` causes different results depending on the execution device.

Observed behavior:

- GPU execution completes without crashing.
- CPU execution terminates with:

```
Floating point exception
```

Expected behavior:

TensorFlow should consistently validate invalid pooling dimensions on both CPU and GPU and raise a framework-level exception such as `InvalidArgumentError` or `ValueError` instead of causing a floating point exception on CPU.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

input_data = tf.constant(
    [[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]],
    dtype=tf.float32,
)

pool = tf.keras.layers.MaxPool1D(
    pool_size=3,
    strides=1,
    padding="valid"
)

y = pool(input_data)

print(y)
```

### Relevant log output

```shell
GPU: No crash

CPU:
Floating point exception
```

**Comment by Ria-K912** [comment]
Hi @TTZTTZ1 @Venkat6871  , I’m interested in working on this issue. Can I take this up?
